application_form/permissions.py

```diff
@@ -0,0 +1,47 @@
+from django.conf import settings
+from django.contrib.auth.models import AnonymousUser
+from rest_framework import authentication, permissions
+from rest_framework.authentication import get_authorization_header
+from rest_framework.exceptions import AuthenticationFailed
+
+
+class DrupalUser(AnonymousUser):
```

Comment:
As we also have actual users coming from Drupal, I think it would be better if this was named so that you cannot mix this up with those. Maybe `DrupalServerUser` ?